ext/opentelemetry-ext-prometheus/tests/test_prometheus_exporter.py

```diff
@@ -0,0 +1,119 @@
+        self._start_http_server_patch = mock.patch(
+            "prometheus_client.exposition._ThreadingSimpleServer",
+            side_effect=self._mock_start_http_server,
+        )
+        self._mock_registry_register = mock.Mock()
+        self._registry_register_patch = mock.patch(
+            "prometheus_client.core.REGISTRY.register",
+            side_effect=self._mock_registry_register,
+        )
+
+    # pylint: disable=protected-access
+    def test_constructor(self):
+        """Test the constructor."""
+        with self._registry_register_patch, self._start_http_server_patch:
+            exporter = PrometheusMetricsExporter(
+                1234, "testaddress", "testprefix"
+            )
+            self.assertEqual(exporter._port, 1234)
+            self.assertEqual(exporter._address, "testaddress")
+            self.assertEqual(exporter._collector._prefix, "testprefix")
+            self.assertTrue(self._mock_start_http_server.called)
+            self.assertTrue(self._mock_registry_register.called)
+
+    def test_shutdown(self):
+        with self._start_http_server_patch, mock.patch(
+            "prometheus_client.core.REGISTRY.unregister"
+        ) as registry_unregister_patch:
+            exporter = PrometheusMetricsExporter()
+            exporter.shutdown()
+            self.assertTrue(registry_unregister_patch.called)
+
+    def test_export(self):
+        with self._registry_register_patch, self._start_http_server_patch:
+            record = MetricRecord(
+                CounterAggregator(), self._test_label_set, self._test_metric
+            )
+            exporter = PrometheusMetricsExporter()
+            result = exporter.export([record])
+            # pylint: disable=protected-access
+            self.assertEqual(len(exporter._collector._metrics_to_export), 1)
+            self.assertIs(result, MetricsExportResult.SUCCESS)
+
+    def test_counter_to_prometheus(self):
+        meter = metrics.Meter()
+        metric = meter.create_metric(
+            "test@name",
+            "testdesc",
+            "unit",
+            int,
+            metrics.Counter,
+            ["environment@"],
+        )
+        kvp = {"environment@": "staging"}
+        label_set = meter.get_label_set(kvp)
+        aggregator = CounterAggregator()
+        aggregator.check_point = 123
+        record = MetricRecord(aggregator, label_set, metric)
+        collector = CustomCollector("testprefix")
+        collector.add_metrics_data([record])
+
+        for prometheus_metric in collector.collect():
+            self.assertEqual(type(prometheus_metric), CounterMetricFamily)
+            self.assertEqual(prometheus_metric.name, "testprefix_test_name")
+            self.assertEqual(prometheus_metric.documentation, "testdesc")
+            self.assertEqual(prometheus_metric.samples[0].value, 123)
+            self.assertEqual(
+                prometheus_metric.samples[0].labels["environment_"], "staging"
+            )
+
```

Comment:
Is it missing to test that a warning is printed when an unsupported metrics type is received?